lib/Core/ExecutorTimers.cpp

```diff
@@ -96,106 +78,92 @@ void Executor::initTimers() {
 
 ///
 
-Executor::Timer::Timer() {}
-
-Executor::Timer::~Timer() {}
-
 void Executor::addTimer(Timer *timer, time::Span rate) {
   timers.push_back(new TimerInfo(timer, rate));
 }
 
 void Executor::processTimers(ExecutionState *current,
                              time::Span maxInstTime) {
-  static unsigned callsWithoutCheck = 0;
-  unsigned ticks = timerTicks;
+  const static time::Span minInterval(TimerInterval);
+  static time::Point lastCheckedTime;
```

Comment:
Please get rid of the static variables -they are globals with their many downsides, i.e. its hard to optimise for the compiler.

(Just a small example: https://godbolt.org/z/zP4jEg )

Reply:
But where should I put it? Into the Executor, into a new TimerContext, into a global :grin:... I tried to keep the structure of that method.

Reply:
;) Everything except anything that smells like a global.

Reply:
This seems like a dangerous conversation, but... :)  This looks to me like an appropriate use of `static` here.  @MartinNowack are you against all such uses in C++ code? ;)

Reply:
@MartinNowack SHould I use Emacs or Vim for the patch?

Reply:
@ccadar No, no - there is `static` and there is `static` ;)

Reply:
@251 Whatever makes you happy!!